### ISSUE TITLE
How does the styling work

### ISSUE THREAD

**Issue by frivoal** [salesforce/standards-explainers#4]
As discussed in the CSSWG call, we need some kind of mechanism to make the styles that you provide via the UA only apply when `appearance` is `base`, as not all these styles are appropriate / compatible with the control when it's either in it's `appearance:auto` or `appearance:none` states. Something like a `:base` pseudo class won't work, as selectors cannot depend on style.

That should at least be acknowledged as an open question.

As for how to answer it…

It was suggested in the CSSWG call that there could maybe be a special stylesheet origin that is somehow only applied when `appearance` is `base`. This seemed somewhat reasonable to me at the time, however:
* that isn't available to authors (including framework authors) who could also want to provide their own styling that only applies when `appearance` is `base` without disturbing the `appearance:auto` or `appearance:none` states.
* What is supposed to happen if you write `appearance:auto` (or `appearance:none`) into that stylesheet? Presumably we can just avoid that question by not doing such a silly thing in the UA stylesheet, but if we make that same mechanism available to authors, we need to deal with that potential loop.

Maybe we could have an `@base-appearance {…}` rule which sets up this special origin (or maybe a special [cascade layer](https://drafts.csswg.org/css-cascade-5/#layering)?), and which would be available both to the UA and to authors to insert the desired styles. It would need to be special in two ways:
* automagically apply when `appearance` is `base`
* ignore the `appearance` property within all declarations blocks that it contains, to prevent loops.

Or maybe something else entirely. Either way, it's an open question, and should be listed as such.

**Comment by gregwhitworth** [comment]
> It was suggested in the CSSWG call that there could maybe be a special stylesheet origin that is somehow only applied when appearance is base. This seemed somewhat reasonable to me at the time, however:

Yes, the origin solution is off the table, it was earlier considered by myself but it wasn't a requirement so I left it off the table.

> that isn't available to authors (including framework authors) who could also want to provide their own styling that only applies when appearance is base without disturbing the appearance:auto or appearance:none states.

Given that the origin isn't in question anymore and they are styles directly in the UA stylesheet, why wouldn't it work as author styles do today? Even in the framework scenario, I'll have my initial styles that when I apply the switch to enable the base DOM & styles I'll adjust. Then the author consuming my framework will extend that with their own styles. So I think this should just work since it will be in the UA stylesheet using the normal cascade.

> What is supposed to happen if you write appearance:auto (or appearance:none) into that stylesheet? Presumably we can just avoid that question by not doing such a silly thing in the UA stylesheet, but if we make that same mechanism available to authors, we need to deal with that potential loop.

I supposed I'm confused here as I'd expect following style computation for there to be a single `appearance` value for the host control/component (eg: `[type=checkbox]`). What am I missing?
